query-node/mappings/src/content/app.ts

```diff
@@ -0,0 +1,108 @@
+import { DatabaseManager, SubstrateEvent } from '@joystream/hydra-common'
+import { ICreateApp, IUpdateApp } from '@joystream/metadata-protobuf'
+import { ChannelId } from '@joystream/types/primitives'
+import { logger } from '@joystream/warthog'
+import { Channel, App } from 'query-node/dist/model'
+import { inconsistentState } from '../common'
+
+export async function processCreateAppMessage(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  channelId: ChannelId,
+  message: ICreateApp
+): Promise<void> {
+  const { name, appMetadata } = message
+  const appId = await createAppId(event)
+
+  const isAppExists = await store.get(App, {
+    where: {
+      name: name,
+    },
+  })
+
+  if (isAppExists) {
+    inconsistentState(`App with this name already exists:`, name)
+  }
+
+  // load channel
+  const channel = await store.get(Channel, {
+    where: { id: channelId.toString() },
+  })
+
+  // ensure channel exists
+  if (!channel) {
+    return inconsistentState('Non-existing channel update requested', channelId)
+  }
+
+  const newApp = new App({
+    name,
+    id: appId,
+    websiteUrl: appMetadata?.websiteUrl || undefined,
+    useUri: appMetadata?.useUri || undefined,
+    smallIcon: appMetadata?.smallIcon || undefined,
+    mediumIcon: appMetadata?.mediumIcon || undefined,
+    bigIcon: appMetadata?.bigIcon || undefined,
+    oneLiner: appMetadata?.oneLiner || undefined,
+    description: appMetadata?.description || undefined,
+    termsOfService: appMetadata?.termsOfService || undefined,
+    platforms: appMetadata?.platforms || undefined,
+    category: appMetadata?.category || undefined,
+    authKey: appMetadata?.authKey || undefined,
+    ownerCuratorGroup: channel.ownerCuratorGroup,
+    ownerMember: channel.ownerMember,
+    channel,
+  })
+  await store.save<App>(newApp)
+  logger.info('App has been created', { name })
+
+  channel.app = newApp
+
+  await store.save<Channel>(channel)
+  logger.info('Channel has been updated', { channel })
```

Comment:
Redundant. Doesn't really need this, other part of the relation will be automatically created.

```suggestion
```